controllers/datafoundation.go

```diff
@@ -171,6 +182,27 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 
 	} else {
+
+		// Checking for CRDs that are installed after the agent was started and add watchers for the new CRDs
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClusterCRDName] {
```

Comment:
Please do not access the underlaying reconciler in the offering code.
You can copy a ref to the AvailableCRDs map into the sub reconciler on init